crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -338,10 +339,21 @@ def prepare_nodes_for_os_upgrade
       end
       if ready_for_reboot
         @logger.debug("Rebooting node #{node.name} for operating system upgrade")
-        node.ssh_cmd("/sbin/reboot")
+        ssh_status = node.ssh_cmd("/sbin/reboot")
+        if ssh_status[0] != 200
+          @logger.error("Upgrade failed for machine #{node.name}. Could not ssh.")
+          upgrade_nodes_failed.push(node.name)
+        end
+      else
+        @logger.error("Upgrade failed for #{node.name}. Node not ready for reboot")
+        upgrade_nodes_failed.push(node.name)
       end
     end
-    true
+    if upgrade_nodes_failed.any?
+      return false, upgrade_nodes_failed
+    else
+      return true
```

Comment:
@dguitarbite Is like in Python (at least in my tests with `irb`):

* `true` is evaluated as `true`
* `[false, [list, of, nodes]]` is evaluated also as `true`

So maybe we can do as in Python: return always `[true, nil]` or `[false, [list, of, nodes]]`, so in the caller side you always use `result, list_of_nodes = object.prepare_nodes_for_os_upgrade(...)`

But I am sure that a Ruby expert can provide a confirmation is this an expected idiom in Ruby too.